TYPO3.Neos/Classes/TYPO3/Neos/Domain/Service/UserService.php

```diff
@@ -214,7 +214,25 @@ public function getCurrentUser()
         if ($this->securityContext->canBeInitialized() === true) {
             $account = $this->securityContext->getAccount();
```

Comment:
This is not introduced with this change, but IMO _that_ is actually the limiting factor here:
`Security\Context::getAccount()` just returns the first authenticated account, whichever the first account is and whatever authentication provider it belongs to.
I think we absolutely have to make this explicit and use `Security\Context::getAccountByAuthenticationProviderName()` instead